test/StockQuoteAnalyzerTest.java

```diff
@@ -41,4 +64,145 @@ public void tearDown() throws Exception {
+        analyzer = new StockQuoteAnalyzer(symbol, generatorMock, audioMock);
+        when(generatorMock.getCurrentQuote()).thenReturn(new StockQuote(symbol, previousClose, lastTrade, change));
+
+        analyzer.playAppropriateAudio();
+        verify(audioMock, times(0)).playHappyMusic();
+        verify(audioMock, times(1)).playErrorMusic();
+        verify(audioMock, times(0)).playSadMusic();
+    }
+
+    @Test (expectedExceptions = InvalidAnalysisState.class)
+    public void getCurrentPriceShouldThrowExceptionWhenWhenNoQuoteReceived() throws Exception {
+        analyzer = new StockQuoteAnalyzer("GM", generatorMock, audioMock);
+        when(generatorMock.getCurrentQuote()).thenReturn(null);
+        analyzer.refresh();
+        analyzer.getCurrentPrice();
+    }
+
+    @Test (dataProvider = "QuoteClasses")
+    public void getCurrentPriceShouldReturnCurrentPricesWhenThereExistsACurrentQuote(String symbol, String name, double previousClose, double lastTrade,  double change) throws Exception{
+        analyzer = new StockQuoteAnalyzer(symbol, generatorMock, audioMock);
+        when(generatorMock.getCurrentQuote()).thenReturn(new StockQuote(symbol, previousClose, lastTrade, change));
+        analyzer.refresh();
+        assertEquals(analyzer.getCurrentPrice(), lastTrade);
+    }
+
+    @Test (expectedExceptions = InvalidAnalysisState.class)
+    public void getChangeSinceCloseShouldThrowExceptionWhenNoQuoteReceived() throws Exception {
+        analyzer = new StockQuoteAnalyzer("GM", generatorMock, audioMock);
+        when(generatorMock.getCurrentQuote()).thenReturn(null);
+        analyzer.refresh();
+        analyzer.getChangeSinceClose();
+    }
+
+    @Test (dataProvider = "QuoteClasses")
+    public void getChangeSinceCloseShouldReturnChangeSinceClose(String symbol, String name, double previousClose, double lastTrade,  double change) throws Exception{
+        analyzer = new StockQuoteAnalyzer(symbol, generatorMock, audioMock);
+        when(generatorMock.getCurrentQuote()).thenReturn(new StockQuote(symbol, previousClose, lastTrade, change));
+        analyzer.refresh();
+        assertEquals(analyzer.getChangeSinceClose(), change);
+    }
+
+    @Test (dataProvider = "TwoChangesQuotes")
+    public void getChangeSinceLastCheckShouldReturnCurrentMinusPreviousPrice(String symbol, String name, double price1, double price2, double price3) throws Exception{
+        analyzer = new StockQuoteAnalyzer(symbol, generatorMock, audioMock);
+        when(generatorMock.getCurrentQuote()).thenReturn(new StockQuote(symbol, price1, price2, price2-price1));
+        analyzer.refresh();
+        when(generatorMock.getCurrentQuote()).thenReturn(new StockQuote(symbol, price2, price3, price3-price2));
+        analyzer.refresh();
+        assertEquals(analyzer.getChangeSinceLastCheck(), (price3-price2));
+    }
+
+    @Test (expectedExceptions = InvalidAnalysisState.class)
```

Comment:
I think it would help if you added in more whitespace to visually organize your tests.  Separate the parts of the test visually withy CRLFs.  It would improve readability.  Also the meaning of variables like price1, 2, and 3 will get lost over time.